iocBoot/templates/st.cmd.template

```diff
@@ -2,6 +2,10 @@
 ################### AUTO-GENERATED DO NOT EDIT ###################
 {%  set production_ioc = production_ioc | int %}
 {%- set project, plc = get_plc_by_name(projects, plc) -%}
+{% if plc.name is not defined %}
+# PLC name invalid or not found in project.
+# Double-check Makefile setting "PLC" and compare with TwinCAT.
+{% else %}
```

Comment:
Does the user see an error of any kind when expanding this template with a bad plc name?

Reply:
I guess this warning comment does make it into the built file

Reply:
I think there was no way to really raise a useful exception in the template, so I went this route.

Without this PR, it raises something like `"name" is not an attribute of None` if the PLC name is set incorrectly.
I'd argue that's more confusing to fail with a confusing exception than generate a non-functional IOC (that should be reviewed)

Perhaps instead we make `pytmc template` add in support for `{% raise_error("PLC name invalid") %}` in the templates

Reply:
I forgot that we can inject our own functions into templates, provided we make them first- maybe that's the best thing to do, because they we can give pointed errors. But maybe that doesn't need to happen unless we have more than a few different errors in mind to show.

I think this is useful as-is fwiw